### ISSUE TITLE
Question about the yolov8l-worldv2 lvis zero shot object detection reproduce.

### ISSUE THREAD

**Issue by minimini-1** [ultralytics/ultralytics#21246]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

> [@KaamComputingNp](https://github.com/KaamComputingNp) for reproducing YOLO-World results on LVIS, please use the LVIS dataset YAML configuration shown in the LVIS dataset documentation. The dataset uses COCO images with different splits and annotations.
> 
> A training example specifically for LVIS with YOLO-World is provided in the YOLO-World documentation under "How do I reproduce the official results of YOLO-World from scratch?" where `val: {"yolo_data": ["lvis.yaml"]}` is specified. Ensure your dataset paths match the YAML structure and consider using the custom `WorldTrainerFromScratch` trainer as shown.

Hi, I have the same question.

I used the weights provided at
[https://docs.ultralytics.com/models/yolo-world/#key-features](https://docs.ultralytics.com/models/yolo-world/#key-features)
and set the class names according to the LVIS dataset documentation. (https://docs.ultralytics.com/ko/datasets/detect/lvis/#sample-images-and-annotations)

However, my results are different from the zero-shot LVIS-minival results shown on the YOLO-World GitHub.
![Image](https://github.com/user-attachments/assets/0a3b3a3a-c1cb-4d47-980d-e8649da7a247)

Does the pre-trained model from Ultralytics use a different pre-training method compared to the one described in the YOLO-World GitHub?
I’d like to understand why I’m unable to reproduce the reported results.

Plus, I also checked relevant issues.
#19928, #19484

**Comment by UltralyticsAssistant** [comment]
👋 Hello @minimini-1, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response 🤖. An Ultralytics engineer will also review your question and assist you as soon as possible!

**Comment by Y-T-G** [comment]
You can try with `rect=False`

**Comment by minimini-1** [comment]
> You can try with `rect=False`

Hi, thank you for your reply.
I try `rect=False`, but the performance is still different from YOLO-World GitHub.

#### output
{'map50_95': 0.1105692718339723, 'map50': 0.14153374449368014, 'map75': 0.11992907645159132}

Do I need to change any settings?
Also, if I evaluate the LVIS-minimal dataset using the weights provided by Ultralytics, should I expect the same performance as reported in the YOLO-World GitHub repository?

**Comment by glenn-jocher** [comment]
The Ultralytics YOLO-World weights may differ from the original YOLO-World paper implementation, which could explain the performance gap you're seeing. The numpy deprecation warning in your output also suggests there might be evaluation compatibility issues affecting the metrics.

For exact reproduction of the original paper results, you'd need to train from scratch using the [WorldTrainerFromScratch trainer](https://docs.ultralytics.com/models/yolo-world/#how-do-i-reproduce-the-official-results-of-yolo-world-from-scratch) with the same datasets (Objects365, GQA, Flickr30k) and evaluation protocol as described in the original work. The Ultralytics implementation provides a convenient interface but may use different training procedures or data preprocessing that impact zero-shot performance on LVIS.

You might also try using the full class names without splitting: `names_list = list(names_dict.values())` instead of taking only the first part before the slash, as the exact class name format can affect detection accuracy.

**Comment by chandan-chakma** [comment]
def main():
    from ultralytics import YOLO
    import yaml

    # Load LVIS class names with UTF-8 encoding
    with open('lvis.yaml', encoding='utf-8') as f:  # Add encoding parameter
        lvis_data = yaml.safe_load(f)
    lvis_classes = list(lvis_data['names'].values())

    model = YOLO("yolov8s-world.pt")
    model.set_classes(lvis_classes)  # Set LVIS classes

    metrics = model.val(
        data='lvis.yaml',
        split='val',
        save_json=True,
        imgsz=640,
        batch=8,
        conf=0.001,
        iou=0.7,
        max_det=1000
    )

if __name__ == "__main__":
    import multiprocessing
    multiprocessing.freeze_support()
    main()

with this config i got simmilar result as paper.

**Comment by glenn-jocher** [comment]
Thanks for sharing that configuration @chandan-chakma! The key differences in your setup that likely improve results are:

- Using `conf=0.001` and `max_det=1000` which matches the evaluation protocol from the original YOLO-World paper
- Using `split='val'` instead of `minival` 
- The full LVIS class names without splitting as I mentioned

@minimini-1 you should try those evaluation parameters, particularly the low confidence threshold which is crucial for zero-shot evaluation where you want to capture all possible detections before applying NMS filtering.
